core/src/toga/widgets/imageview.py

```diff
@@ -72,26 +70,21 @@ def rehint_imageview(image, style, scale=1):
 class ImageView(Widget):
     def __init__(
         self,
-        image: str
-        | Path
-        | bytes
-        | bytearray
-        | memoryview
-        | PIL.Image.Image
-        | None = None,
+        image: str | Path | bytes | bytearray | memoryview | Any | None = None,
         id=None,
         style=None,
     ):
         """
         Create a new image view.
 
-        :param image: The image to display. This can take all the same formats as the
-            `src` parameter to :class:`toga.Image` -- namely, a file path (as string
-            or :any:`pathlib.Path`), bytes data in a supported image format,
-            or :any:`PIL.Image.Image`.
+        :param image: The image to display. This can take all the same formats
+            as the `src` parameter to :class:`toga.Image` -- namely, a file path
+            (as string or :any:`pathlib.Path`), bytes data in a supported image
+            format, an instance of the platform's native Image type, or
```

Comment:
Although, if this is now part of the public API we should probably list the native types explicitly, rather than making the reader guess.

Reply:
Oh, I see them now, they're on the `Image` page, but not referenced from the constructor docstring. If we can have an `ImageType` alias linked from both `Image` and `ImageView`, that would be a good place to move that list.

The list of  file formats (PNG, JPG, etc.) should probably stay where it is, because they apply to saving images as well as loading them. But there should be a link to that list from each relevant place.

Reply:
I was on the fence about documenting the internal types at all. As an end user, unless you're reaching into the native layer to implement a feature, you shouldn't need to know the internal types. On that basis, I'm OK with this being a *little* obscure. 

As mentioned in the other comment about types, I'm not aware of any way that we can structure this in Sphinx that is readable *and* mollifies type checkers; including not being able to document the actual type. I'm open to any suggestions.

Reply:
> unless you're reaching into the native layer to implement a feature, you shouldn't need to know the internal types

We recommend people to reach into the native layer on a fairly regular basis, so we should document this properly. Saying "Can also accept the platform's native image format" without clearly indicating what that means is just going to annoy the reader. A link to the list would be enough to fix this.

Reply:
Following the type documentation wrangling I did for https://github.com/beeware/toga/pull/2259, I've adopted the same typing and documentation pattern for Image. This means there's some overlap in the patch (e.g., the updates to conf.py).